### PR TITLE
Add a check to only to encrypt the password field when a password is …

### PR DESCRIPTION
**Pull request by weyert**
…passed through the values-object of the hook

### CODE REVIEW

File: src/authentication/templates/api/models/User.js [hunk 1]
```diff
@@ -67,6 +68,8 @@ export default {
   },
 
   beforeCreate(values, next) {
+    if (false === values.hasOwnProperty('password')) return next()
```
Comment (ghaiklor) [inline src/authentication/templates/api/models/User.js]: Don't forgot to add semi-colon after `next()` :+1:

File: src/authentication/templates/api/models/User.js [hunk 2]
```diff
@@ -56,6 +56,7 @@ export default {
   },
 
   beforeUpdate(values, next) {
+    if (false === values.hasOwnProperty('password')) return next()
```
Comment (ghaiklor) [inline src/authentication/templates/api/models/User.js]: Also why do you set `false` when you can just write `!values.hasOwnProperty()`

Comment (weyert) [inline src/authentication/templates/api/models/User.js]: Mainly for readability more clear what we are checking against and putting it on the left side avoids accidentally using =

**Comment by weyert** [comment]
Added the semicolon to the respective lines

**Pull request merged**